source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRView.mm

```diff
@@ -593,6 +702,28 @@ - (void)observeValueForKeyPath:(NSString *)path ofObject:(id)object change:(NSDi
     }
 }
 
+- (void)loadBackgroundImageAccordingToResourceResolverIF:(std::shared_ptr<BackgroundImage> const &)backgroundImage key:(NSString *)key observerAction:(ObserverActionBlock)observerAction
```

Comment:
I should have made loadBackgroundImageAccordingToResourceResolverIF to take common object-c object that handles generating hash key and url, then you wouldn't have to write this new function. I can add it after your commit, or I can add it to your branch.

Reply:
Well, `loadBackgroundImage...` and `loadImage...` are practically the same thing. The reason I created a new one is because the `BackgroundImage` object is not a `BaseCardElement`. If there's a way to allow `BackgroundImage` and a `BaseCardElement` to be passed in as the same parameter, I think that might solve this problem.

In other words, feel free to add it to this branch haha.